src/main/java/net/glowstone/net/handler/legacyping/LegacyPingHandler.java

```diff
@@ -0,0 +1,101 @@
+                        sendByteBuf(channelHandlerContext, responseToByteBuf(String.format("%s\u00a7%d\u00a7%d", legacyPingEvent.getMotd(), legacyPingEvent.getNumPlayers(), legacyPingEvent.getMaxPlayers())));
+                        legacyPingProtocol = true;
+                        break;
+                    case 1:
+                        if (bytebuf.readByte() == (byte) 0x01) {
+                            sendByteBuf(channelHandlerContext, responseToByteBuf(String.format("\u00a71\u0000%d\u0000%s\u0000%s\u0000%d\u0000%d", GlowServer.PROTOCOL_VERSION, GlowServer.GAME_VERSION, legacyPingEvent.getMotd(), legacyPingEvent.getNumPlayers(), legacyPingEvent.getMaxPlayers())));
+                            legacyPingProtocol = true;
+                        }
+                        break;
+                    default:
+                        if (bytebuf.readByte() == (byte) 0x01 && bytebuf.readByte() == (byte) 0xFA && "MC|PingHost".equals(new String(bytebuf.readBytes(bytebuf.readShort() * 2).array(), Charsets.UTF_16BE))) {
+                            int dataLength = bytebuf.readUnsignedShort();
+                            short clientVersion = bytebuf.readUnsignedByte();
+                            String hostname = bytebuf.readBytes(bytebuf.readShort() * 2).toString(Charsets.UTF_16BE);
+                            @SuppressWarnings("unused")
+                            int port = bytebuf.readInt();
+
+                            if (clientVersion >= 73 && 7 + (hostname.length() * 2) == dataLength && bytebuf.readableBytes() == 0) {
+                                sendByteBuf(channelHandlerContext, responseToByteBuf(String.format("\u00a71\u0000%d\u0000%s\u0000%s\u0000%d\u0000%d", GlowServer.PROTOCOL_VERSION, GlowServer.GAME_VERSION, legacyPingEvent.getMotd(), legacyPingEvent.getNumPlayers(), legacyPingEvent.getMaxPlayers())));
+                                legacyPingProtocol = true;
+                            }
+                        }
+                        break;
+                }
+            }
+        } catch (RuntimeException e) {
+            // Silently catch the exception
+            ;
+        } finally {
+            // check if not successful, otherwise the connection has already been closed
+            if (!legacyPingProtocol) {
+                bytebuf.resetReaderIndex();
+                channelHandlerContext.pipeline().remove("legacy_ping");
+                channelHandlerContext.fireChannelRead(object);
+            }
+        }
+    }
+
+    private void sendByteBuf(ChannelHandlerContext channelhandlercontext, ByteBuf bytebuf) {
+        channelhandlercontext.writeAndFlush(bytebuf).addListener(ChannelFutureListener.CLOSE);
+    }
+
+    private ByteBuf responseToByteBuf(String string) {
+        ByteBuf bytebuf = Unpooled.buffer();
+
+        bytebuf.writeByte(0xFF);
+        bytebuf.writeShort(string.length());
+        try {
+            bytebuf.writeBytes(string.getBytes("UTF-16BE"));
```

Comment:
They aren't equal you're right but that's not what the specification does. It sends the number of characters not the length of the data.